In [14]:
import pandas as pd
import numpy as np
import pyreadstat
import glob
import os

In [15]:
anes_24_raw_path = 'data/raw/anes/anes-2024.csv'
cces_24_raw_path = 'data/raw/cces/cces-2024.csv'
igs_24_raw_path = 'data/raw/igs/igs-2024.csv'
rA24 = pd.read_csv(anes_24_raw_path) # check that it comes unweighted 
rC24 = pd.read_csv(cces_24_raw_path) # check that it comes unweighted 
rI24 = pd.read_csv(igs_24_raw_path)

/var/folders/67/0s5rn4vn3sd9vsmn400fmxr00000gn/T/ipykernel_29773/143579392.py:4: DtypeWarning: Columns (8,10,11,16,18,19,20,22,23,24,26,27,28,30,31,36,38,39,1448,1449,1450,1451,1452,1455,1456,1457,1458,1459,1463,1464,1465) have mixed types. Specify dtype option on import or set low_memory=False.
  rA24 = pd.read_csv(anes_24_raw_path) # check that it comes unweighted


In [16]:
all_datapoints_count = 0

num_rA24 = len(rA24[rA24['V241529'] == 6])
num_rC24 = len(rC24[rC24['inputstate']==6])
num_rI24 = len(rI24)

all_datapoints_count += num_rA24 + num_rC24 + num_rI24
print(f"Number of datapoints in ANES: {num_rA24}")
print(f"Number of datapoints in CCES: {num_rC24}")
print(f"Number of datapoints in IGS: {num_rI24}")

Number of datapoints in ANES: 457
Number of datapoints in CCES: 4900
Number of datapoints in IGS: 6536


In [17]:
ppic_folder = 'data/raw/ppic/'

rPList24 = {}
num_rP24 = 0
for file_path in glob.glob(os.path.join(ppic_folder, "*.sav")):
    print(f"Opening {file_path}...")
    df, meta = pyreadstat.read_sav(file_path)
    rPList24[os.path.basename(file_path)] = { # extract file name with os.path.basename(file_path)
        'df':df, 
        'meta':meta
        }
    num_rP24 += len(df)
    print(f"Loaded {len(df)} rows, {len(df.columns)} columns")

all_datapoints_count += num_rP24
print(f"Total number of datapoints in PPIC: {num_rP24}")

Opening data/raw/ppic/2024.09.18.release.sav...
Loaded 1605 rows, 101 columns
Opening data/raw/ppic/2024.10.23.release.sav...
Loaded 1646 rows, 70 columns
Opening data/raw/ppic/2024.07.17.release.sav...
Loaded 1648 rows, 100 columns
Opening data/raw/ppic/2024.12.09.release.sav...
Loaded 2344 rows, 114 columns
Opening data/raw/ppic/2024.06.13.release.sav...
Loaded 1677 rows, 67 columns
Opening data/raw/ppic/2024.04.10.release.sav...
Loaded 1605 rows, 107 columns
Opening data/raw/ppic/2024.02.22.release.sav...
Loaded 1628 rows, 103 columns
Total number of datapoints in PPIC: 12153


In [18]:
print(f"All datapoints so far: {all_datapoints_count}")

All datapoints so far: 24046


# From Documentation
- CCES: 4900 Californians, 1009 Oregon, 1382 Washington --> 7291
- ANES: 457 Californians, 87 Oregon, 85 Washington --> 629 

## ANES

In [19]:
print("California: ", len(rA24[rA24['V241529'] == 6]))
print("Oregon: ", len(rA24[rA24['V241529'] == 41]))
print("Washington: ", len(rA24[rA24['V241529'] == 53]))

California:  457
Oregon:  87
Washington:  85


In [ ]:
len(rA24[(rA24['V241529'] == 6) | (rA24['V241529'] == 53) | (rA24['V241529'] == 41)])


629